src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -1,145 +1,94 @@
+import static com.wikia.webdriver.common.core.configuration.Configuration.getEnv;
+import static com.wikia.webdriver.common.core.configuration.Configuration.getEnvType;
+import static com.wikia.webdriver.common.core.configuration.Configuration.getForceHttps;
+import static com.wikia.webdriver.common.core.configuration.Configuration.getForceLanguageInPath;
+import static com.wikia.webdriver.common.core.configuration.Configuration.getWikiLanguage;
+import static com.wikia.webdriver.common.core.configuration.Configuration.getWikiName;
+
 import com.wikia.webdriver.common.core.configuration.EnvType;
 import okhttp3.HttpUrl;
-import org.apache.commons.lang.StringUtils;
 import org.openqa.selenium.WebDriverException;
 
-public class UrlBuilder {
+public class UrlBuilder extends BaseUrlBuilder {
 
 
-  protected String env;
-  protected EnvType envType;
+  private final String wikiName;
   private Boolean forceHttps;
   private Boolean forceLanguageInPath;
-  protected static final String HTTP_PREFIX = "http://";
+  private String language;
 
-  public UrlBuilder() {
-    this.env = Configuration.getEnv();
-    this.envType = Configuration.getEnvType(this.env);
-    this.forceHttps = Configuration.getForceHttps();
-    this.forceLanguageInPath = Configuration.getForceLanguageInPath();
+  private UrlBuilder(String wiki, String env, Boolean forceHttps, Boolean forceLanguageInPath, String language) {
+    super(env);
+    this.wikiName = wiki;
+    this.forceHttps = forceHttps;
+    this.forceLanguageInPath = forceLanguageInPath;
+    this.language = language;
   }
 
-  public UrlBuilder(String env, Boolean forceHttps) {
-    this.env = env;
-    this.envType = Configuration.getEnvType(this.env);
-    this.forceHttps = forceHttps;
-    this.forceLanguageInPath = Configuration.getForceLanguageInPath();
+  public static UrlBuilder createUrlBuilder() {
+    return createUrlBuilderForWikiAndLang(getWikiName(), getWikiLanguage());
   }
 
-  public UrlBuilder(String env, Boolean forceHttps, Boolean forceLanguageInPath) {
-    this.env = env;
-    this.envType = Configuration.getEnvType(this.env);
-    this.forceHttps = forceHttps;
-    this.forceLanguageInPath = forceLanguageInPath;
+  public static UrlBuilder createUrlBuilderForWiki(String wiki) {
+    return createUrlBuilderForWikiAndLang(wiki, getWikiLanguage());
   }
 
-  public String getLanguageForWiki() {
-    return Configuration.getWikiLanguage() == null ? Configuration.DEFAULT_LANGUAGE : Configuration.getWikiLanguage();
+  public static UrlBuilder createUrlBuilderForWikiAndLang(String wiki, String language) {
+    return new UrlBuilder(wiki, getEnv(), getForceHttps(), getForceLanguageInPath(), language);
   }
 
   public String normalizePageName(String pageName) {
     return pageName.replace(" ", "_");
   }
 
   public String getUrlForPageWithWWW(String pageName) {
-    return getUrlForWiki(Configuration.getWikiName(), true) + pageName;
+    return getUrlForWiki(getWikiName(), true) + pageName;
```

Comment:
as far as I'm seeing below, this is the function declaration: ```public String getUrlForWiki(String language, boolean addWWW)```
So the first parameter is a language, not a wiki name.

Also I'm not a big fan of having several `getUrlForWiki` that require different sets of parameters. Especially the one without params - `getUrlForWiki()` suggests I'm supposed to pass a wiki as a param. Can we do renames, for example:
- `getUrlForWikiPage(String pageName)` -> `getUrlForPage(String pageName)`
- `getUrlForWiki()` -> `getUrl()`
- `getUrlForWiki(String language, boolean addWWW)` -> `getUrlForLanguage(String language, boolean addWWW)`